tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1566,11 +1631,15 @@ Status Converter::BuildCudaEngine(
     return static_cast<int32_t>(trt_algorithm_id);
   }();
 
+  std::unique_ptr<nvinfer1::IAlgorithmSelector> trt_algorithm_selector;
   if (trt_algorithm_id >= 0) {
     VLOG(1) << "Forcing TRT algorithm selection to: ID=" << trt_algorithm_id;
-    StaticAlgorithmSelector trt_algorithm_selector(trt_algorithm_id);
-    builder_config->setAlgorithmSelector(&trt_algorithm_selector);
+    trt_algorithm_selector.reset(new StaticAlgorithmSelector(trt_algorithm_id));
+  }
+  else {
+    trt_algorithm_selector.reset(new FlakyAlgorithmRejector());
   }
+  builder_config->setAlgorithmSelector(trt_algorithm_selector.get());
```

Comment:
Another reason for merging the two selectors into one: the code here won't allow you to filter the bad SHUFFLE implementation when you use the environment for static tactics selection, which is not what we want to see.